### DISCUSSION TITLE
2023-06-22 Community Call

### DISCUSSION THREAD

**Discussion by pshriwise** [svalinn/DAGMC#883] [Community Calls]
Attendees:

UW: @gonuke 
ANL: @pshriwise 
ORNL: Ahmad Ibrahim, Kyle Grammer
First Light Fusion: @shimwell, @rlbarker
Copenhagen Atomics: @ebknudsen
TUV: Lewis MacFarlane

**Comment by gonuke** [comment]
PyNE Updates:
* ORNL/SNS has been struggling through a PyNE installation
* UW team has been [similarly updating the installation script and improving long term robustness](https://github.com/pyne/pyne/pull/1477)

**Reply by KBGrammer** [reply]
The main thing we learned from that is that numpy must be later than 1.19.5 and earlier than 1.24.0. Numpy 1.20.0 is when they deprecated np.int and other similar types that were replaced by native python types. Numpy 1.24.0 (I think) is when those deprecations expired and no longer supported at all. I believe the fix requires a lot of code changes to replace np.int, np.int64, np.in32, np.bool, etc. with native python types.

Another issue we haven't solved is that we can't build nuc_data.h5 against our data directory because the following error appears:

`~/$nuc_data_make --datapath ${DATAPATH} -m all -o testme.h5`
`Making nuc_data at testme.h5`
`skipping atomic mass data table creation; already exists.`
`skipping scattering lengths data table creation; already exists.`
`skipping ENSDF decay data table creation; already exists.`
`skipping simple XS data table creation; already exists.`
`Adding cinder data...`
` energy group boundaries.`
` neutron absorption cross sections.`
`Traceback (most recent call last):`
` File "${PYNEDIR}/bin/nuc_data_make", line 4, in <module>`
`   __import__('pkg_resources').run_script('pyne==0.7.7', 'nuc_data_make')`
` File "${PYTHONDIR}/lib/python3.9/site-packages/pkg_resources/__init__.py", line 720, in run_script`
`   self.require(requires)[0].run_script(script_name, ns)`
` File "${PYTHONDIR}/lib/python3.9/site-packages/pkg_resources/__init__.py", line 1559, in run_script`
`   exec(code, namespace, namespace)`
` File "${PYNEDIR}/lib/python3.9/site-packages/pyne-0.7.7-py3.9.egg/EGG-INFO/scripts/nuc_data_make", line 6, in <module>`
`   main()`
` File "${PYNEDIR}/lib/python3.9/site-packages/pyne-0.7.7-py3.9.egg/pyne/dbgen/nuc_data_make.py", line 205, in main`
`   make_map[mo](args)`
` File "${PYNEDIR}/lib/python3.9/site-packages/pyne-0.7.7-py3.9.egg/pyne/dbgen/cinder.py", line 910, in make_cinder`
`   make_mg_absorption(nuc_data, build_dir)`
` File "${PYNEDIR}/lib/python3.9/site-packages/pyne-0.7.7-py3.9.egg/pyne/dbgen/cinder.py", line 247, in make_mg_absorption`
`   absorption_table = db.create_table(`
` File "${PYTHONDIR}/lib/python3.9/site-packages/tables/file.py", line 1022, in create_table`
`   ptobj = Table(parentnode, name,`
` File "${PYTHONDIR}/lib/python3.9/site-packages/tables/table.py", line 807, in __init__`
`   super().__init__(parentnode, name, new, filters, byteorder, _log,`
` File "${PYTHONDIR}/lib/python3.9/site-packages/tables/leaf.py", line 260, in __init__`
`   super().__init__(parentnode, name, _log)`
` File "${PYTHONDIR}/lib/python3.9/site-packages/tables/node.py", line 249, in __init__`
`   self._v_objectid = self._g_create()`
` File "${PYTHONDIR}/lib/python3.9/site-packages/tables/table.py", line 982, in _g_create`
`   self._v_chunkshape = self._calc_chunkshape(`
` File "${PYTHONDIR}/lib/python3.9/site-packages/tables/leaf.py", line 321, in _calc_chunkshape`
`   complib = self.filters.complib`
`AttributeError: 'numpy.int64' object has no attribute 'complib'`
`Closing remaining open files:testme.h5...done`

**Reply by gonuke** [reply]
This appears to be a compatabilty issue between pytables and numpy.

**Comment by shimwell** [comment]
cadquery updates

two issues related to dagmc
https://github.com/CadQuery/cadquery/issues/1346 round trip keeping track of volumes
https://github.com/CadQuery/cadquery/issues/1243 gmsh integration

two pull requests related to dagmc
https://github.com/CadQuery/cadquery/pull/1353 adds imprinting and volume tracking
https://github.com/CadQuery/cadquery/pull/1343 introduces _address for in memory use

**Comment by pshriwise** [comment]
Shared surface discussion:

@ebknudsen made a couple of steps (leaps??) forward:
 
  - the realization that using STLS files representing each _surface_ (as opposed to volume) of the model allows us to potentially handle coincident parts
  - discovered a way to insert shared surfaces into a `.h5m` file from CADQuery using the hashes of surfaces to determine whether or not the surface has already been meshed (tessellated). The surface's senses seem to be correct, but additional testing is underway.

The potential end result is that we would be able to represent shared/merged surfaces coming from CADQuery 🎉 

Thanks for the relevant links above @shimwell!

**Comment by pshriwise** [comment]
Notes on improvements to the current overlap check tool: #797
